app/domain/authentication/security/process_request_proxies.rb

```diff
@@ -0,0 +1,58 @@
+require 'command_class'
+
+module Authentication
+  module Security
+    ProcessRequestProxies ||= CommandClass.new(
+      dependencies: {
+        # InputValidation
+      },
+      inputs: [:request]
+    ) do
+
+      def call
+        validate_request
+        process_xff_header
+      end
+
+      private
+
+      Struct.new('RemoteAddress', :client_ip, :proxy_list)
+
+      def validate_request
+        raise ArgumentError, '@request' if @request.nil?
+        raise ArgumentError, '@request.remote_addr' if nil_or_empty @request.remote_addr
```

Comment:
don't we want to raise our own error?

Reply:
This is not likely to happen. I consider it a `RuntimeExcpeiton` from which the user cannot recover.

Reply:
i am not a fan of this check if the scenario doesn't really exist.

Reply:
and if it _does_ happen i would rather have a meaningful log message than an `ArgumentError`